base_export_async/models/delay_export.py

```diff
@@ -118,37 +111,28 @@ def export(self, params):
             date_today + relativedelta(days=+int(time_to_live))
         )
 
-        # TODO : move to email template
         odoo_bot = self.sudo().env.ref("base.partner_root")
         email_from = odoo_bot.email
         model_description = self.env[model_name]._description
-        self.env["mail.mail"].create(
+
+        export_record.write(
             {
+                "url": url,
+                "expiration_date": expiration_date,
+                "model_description": model_description,
+            }
+        )
+
+        self.env.ref("base_export_async.delay_export_mail_template").send_mail(
+            export_record.id,
+            email_values={
                 "email_from": email_from,
                 "reply_to": email_from,
                 "recipient_ids": [(6, 0, users.mapped("partner_id").ids)],
-                "subject": _("Export {} {}").format(
-                    model_description, fields.Date.to_string(fields.Date.today())
-                ),
-                "body_html": _(
-                    """
-                <p>Your export is available <a href="{}">here</a>.</p>
-                <p>It will be automatically deleted the {}.</p>
-                <p>&nbsp;</p>
-                <p><span style="color: #808080;">
-                This is an automated message please do not reply.
-                </span></p>
-                """
-                ).format(url, expiration_date),
-                "auto_delete": True,
-            }
+            },
         )
 
     @api.model
     def cron_delete(self):
-        time_to_live = (
-            self.env["ir.config_parameter"].sudo().get_param("attachment.ttl", 7)
-        )
         date_today = fields.Date.today()
-        date_to_delete = date_today + relativedelta(days=-int(time_to_live))
-        self.search([("create_date", "<=", date_to_delete)]).unlink()
+        self.search([("expiration_date", "<=", date_today)]).unlink()
```

Comment:
Given that the expiration_date field has no default value, it reads like this refactoring could potentially leave the exports on the server indefinitely.